notional-finance/contracts-v2
# sherlock audit 2023 03 notional judging 196

- vault: removing transfers from settlement
- vaults: moving function to reduce bytecode size
- vaults: refactoring vault cash emit
- vaults: adding excess secondary cash liquidation
- vaults: fix bug in update
- vaults: fix event in secondary borrow
- vaults: enabling withdraws of excess cash
- fix: changing cash group precision
- vault: changing sign error in settlement
- tests: updating internal tests
- vaults: fixes for cash settlement and liquidation
- tests: vault cash settle and liquidation
- fix: edge case in vault events
- events: removing unused events
- tests: removing some flaky conditions
- test: adding event tests
- events: removing unused events
- tests: first round of fixes for events
- tests: updating internal tests
- tests: updates to vault tests
- events: rearranging events
- events: fixing secondary deleverage events
- tests: secondary deleverage tests
- events: updates to bundles
- test: fixing proxy test
- fix: compiling rebalancing into treasury action
- fix: update governance and init market action
- events: remove deprecated events
- docs: add comment to vault liquidation action
- fix: deployment issue
- tests: fix event parsing
- events: fixing event parsing
- deploy: initial commit of configuration
- adding arbitrum config
- arb: updating arb deployment script
- updating chainlink adapter
- updating parameters
- adding supply cap checks
- updating configuration
- deploy: adding chain id check
- events: reducing gas usage of event
- tests: fixing local testing for beacons
- test: fix vault roll test
- deploy: saving changes before deploy
- updating abi
- adding arbitrum addresses
- updating inspect
- updating inspect file
- fix swapped symbols
- fix: sherlock-audit/2023-03-notional-judging#12 incorrect toUint72
- fix: sherlock-audit/2023-03-notional-judging#129 adding revert in case of negative liquidation factor
- fix: sherlock-audit/2023-03-notional-judging#180 prevent fee on transfer secondary currencies
- fix: sherlock-audit/2023-03-notional-judging#184 check secondary borrow on exit
- fix: sherlock-audit/2023-03-notional-judging#183 fetch prime rate during liquidation
- fix: sherlock-audit/2023-03-notional-judging#204 sherlock-audit/2023-03-notional-judging#194 incorrect inequality in deleverage vault
- fix: sherlock-audit/2023-03-notional-judging#196 fix lend at zero settlement events
